File: 0011_container_with_most_water/container_water.js
/**
 * Time:    O(n)
 * Space:   O(1)
 * @param   {number[]} height
 * @return  {number}
 */
const maxArea = height => {
  let l = 0,
    r = height.length - 1
  let max = Number.MIN_SAFE_INTEGER
  while (l < r) {
    let min = Math.min(height[l], height[r])
    max = Math.max(max, min * (r - l))
    height[r] > height[l] ? l++ : r--
  }
  return max
}

File: 0011_container_with_most_water/container_water.rb
def max_area(height)
    # Time: O(n)
    # Space: O(1)
    max = 0
    i, j = 0, height.length - 1
    while i < j do
        min = [height[i], height[j]].min
        max = [max, (j - i) * min].max
        if height[i] > height[j]
            j -=1
        else
            i += 1
        end
    end
    max
end